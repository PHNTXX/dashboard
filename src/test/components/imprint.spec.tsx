import { fireEvent, render } from "@testing-library/react";
import Imprint, {
  IImprintProps,
  ImprintField,
  onClose,
} from "../../components/imprint";

const props: IImprintProps = {
  name: {
    text: "Test Name",
    link: "#",
  },
  address: {
    text: "Test Address",
    link: "#",
  },
  phone: {
    text: "Test Phone",
    link: "#",
  },
  email: {
    text: "Test Email",
    link: "#",
  },
  url: {
    text: "Test URL",
    link: "#",
  },
  text: "This is a test!",
};

describe("imprint.tsx", () => {
  const location: Location = window.location;

  beforeEach(() => {
    // @ts-ignore
    delete window.location;

    window.location = {
      ...location,
      reload: jest.fn(),
    };
  });

  it("Tests Imprint", () => {
    const { asFragment } = render(<Imprint imprint={props} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("Tests onClose with #imprint", () => {
    const location = window.location.href;
    window.location.href = location + "#imprint";
    onClose();
    expect(window.location.href).toEqual(location);
  });

  it("Tests onClose without #imprint", () => {
    const location = window.location.href;
    onClose();
    expect(window.location.href).toEqual(location);
  });

  it("Tests ImprintField", () => {
    const { asFragment } = render(<ImprintField field={props.name} />);
    expect(asFragment).toMatchSnapshot();
  });
});
