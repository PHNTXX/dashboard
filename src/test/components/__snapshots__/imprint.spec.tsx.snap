// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imprint.tsx Tests Imprint 1`] = `[Function]`;

exports[`imprint.tsx Tests ImprintField 1`] = `[Function]`;
